### PR TITLE
72 user story as user i want to configure per harvest source i have want to harvest the catalog as well

### PR DESCRIPTION
**Pull request by hcvdwerf**
## 🚀 Pull Request Checklist

- **Title:** 
  - `[ ]` A brief, descriptive title for the changes.

- **Description:**
  - `[ ]` Provide a clear and concise description of your pull request, including the purpose of the changes and the approach you've taken.

- **Context:**
  - `[ ]` Why are these changes necessary? What problem do they solve? Link any related issues.

- **Changes:**
  - `[ ]` List the major changes you've made, ideally organized by commit or feature.

- **Testing:**
  - `[ ]` Describe how the changes have been tested. Include any relevant details about the testing environment and the test cases.

- **Screenshots (if applicable):**
  - `[ ]` If your changes are visual, include screenshots to help explain your changes.

- **Additional Information:**
  - `[ ]` Add any other information that might be useful for reviewers, such as considerations, discussions, or dependencies.

- **Checklist:**
  - `[ ]` I have checked that my code adheres to the project's style guidelines and that my code is well-commented.
  - `[ ]` I have performed self-review of my own code and corrected any misspellings.
  - `[ ]` I have made corresponding changes to the documentation (if applicable).
  - `[ ]` My changes generate no new warnings or errors.
  - `[ ]` I have added tests that prove my fix is effective or that my feature works.
  - `[ ]` New and existing unit tests pass locally with my changes.

## Summary by Sourcery

Add support for configurable catalog harvesting in FairDataPointRecordProvider and update FairDataPointCivityHarvester to utilize this configuration. Enhance logging and update documentation to reflect the new configuration option. Include new unit tests to ensure the feature works as expected.

New Features:
- Introduce a configuration option to enable or disable catalog harvesting in the FairDataPointRecordProvider class.

Enhancements:
- Refactor FairDataPointCivityHarvester to support the new catalog harvesting configuration.
- Update logging format for better readability and consistency.

Documentation:
- Add documentation for the new `ckanext.fairdatapoint.harvest_catalogs` configuration setting in the README.md.

Tests:
- Add unit tests for the new catalog harvesting feature in FairDataPointRecordProvider.
- Add unit tests for the FairDataPointCivityHarvester to verify the new configuration handling.

### CODE REVIEW

**Comment by Markus92** [comment]
@sourcery-ai review

File: ckanext/fairdatapoint/tests/test_record_provider.py [hunk 1]
```diff
@@ -105,3 +105,27 @@ def test_orcid_call(self, mocker):
...
+        mocker.patch("ckanext.fairdatapoint.harvesters.domain.fair_data_point.FairDataPoint.get_graph",
+                     new=fdp_get_graph)
+        fdp_get_graph.return_value = Graph().parse(Path(TEST_DATA_DIRECTORY, "process_catalogs.ttl"))
+
+        self.fdp_record_provider.harvest_catalogs = harvest_catalogs
+        actual_result = self.fdp_record_provider._process_catalog("http://example.com/catalog1")
+
+        # Assertions
+        for key, value in expected_results.items():
+            assert actual_result[key] == value
```
Comment (PatrickDekkerHealthRI) [inline ckanext/fairdatapoint/tests/test_record_provider.py:131]: Do we need to check, if there are no additional entries in actual_result ?

**Review by PatrickDekkerHealthRI — COMMENTED** [review]
Please see comment in code

File: ckanext/fairdatapoint/tests/test_record_provider.py (hunk 1 above)
Comment (Markus92) [reply ckanext/fairdatapoint/tests/test_record_provider.py:131]: Could do with the == operator if we don't want additional entries.

Comment (hcvdwerf) [reply ckanext/fairdatapoint/tests/test_record_provider.py]: aangepast

**Review by PatrickDekkerHealthRI — APPROVED** [review]
Nice to have some proper unit testing!

**Pull request merged**